assembl/lib/piwik.py

```diff
@@ -0,0 +1,197 @@
+    params = {
+        "module": "API",
+        "format": "JSON",
+        "token_auth": piwik_api_token
+    }
+    params["method"] = "SitesManager.getSitesIdFromSiteUrl"
+    params["url"] = url
+    result = requests.get(piwik_url, params=params, timeout=15)
+
+    if result.status_code != 200:
+        raise requests.ConnectionError()
+    content = result.json() # Content should be either an empty array, or an array like [{"idsite":"44"}]
+    # print "piwik_SitesManager_getSitesIdFromSiteUrl", content
+    if not isinstance(content, list):
+        raise requests.ConnectionError()
+
+    return content
+
+
+def piwik_SitesManager_addSite(piwik_url, piwik_api_token, siteName, urls, ecommerce = '', siteSearch = '', searchKeywordParameters = '', searchCategoryParameters = '', excludedIps = '', excludedQueryParameters = '', timezone = '', currency = '', group = '', startDate = '', excludedUserAgents = '', keepURLFragments = '', param_type = '', settings = '', excludeUnknownUrls = ''):
+    """Returns the Piwik id_site of the created website
+    Warning: A new Piwik site is created everytime this method is called, even if another Piwik website already exists with the same name and URLs"""
+    params = {
+        "module": "API",
+        "format": "JSON",
+        "token_auth": piwik_api_token
+    }
+    params["method"] = "SitesManager.addSite"
+    params["siteName"] = siteName
+    params["urls"] = urls
+    if ( ecommerce ):
+        params["ecommerce"] = ecommerce
+    if ( siteSearch ):
+        params["siteSearch"] = siteSearch
+    if ( searchKeywordParameters ):
+        params["searchKeywordParameters"] = searchKeywordParameters
+    if ( searchCategoryParameters ):
+        params["searchCategoryParameters"] = searchCategoryParameters
+    if ( excludedIps ):
+        params["excludedIps"] = excludedIps
+    if ( excludedQueryParameters ):
+        params["excludedQueryParameters"] = excludedQueryParameters
+    if ( timezone ):
+        params["timezone"] = timezone
+    if ( currency ):
+        params["currency"] = currency
+    if ( group ):
+        params["group"] = group
+    if ( startDate ):
+        params["startDate"] = startDate
+    if ( excludedUserAgents ):
+        params["excludedUserAgents"] = excludedUserAgents
+    if ( keepURLFragments ):
+        params["keepURLFragments"] = keepURLFragments
+    if ( param_type ):
+        params["type"] = param_type
+    if ( settings ):
+        params["settings"] = settings
+    if ( excludeUnknownUrls ):
+        params["excludeUnknownUrls"] = excludeUnknownUrls
+
+    result = requests.get(piwik_url, params=params, timeout=15)
+
+    if result.status_code != 200:
+        raise requests.ConnectionError()
+
+    content = result.json() # Content should be something like {"value": 47}
+    # print "piwik_SitesManager_addSite", content
+
+    if not content:
+        raise requests.ConnectionError()
+
+    if "value" in content:
```

Comment:
it can be replaced by: `return content.get("value", False)`

Reply:
oh yes sometimes I forget the existence of this

Reply:
improvement done